### PR TITLE
some fixes and additions

### PR DESCRIPTION
**Pull request by wabkia**
in my quest to add a defense when attacked in fullauto mode, i had to move some things around. the main issue seems to be that sometimes the bot just gets stuck in some state, but that was happening before i made changes :P i think fishing hole detection overall was improved but it also increased the chance that the bot will move on from a pond that it has already cast in.  on a guess i added some rotation when the QR is 'none' and it hasnt stopped because it 'cant find QR' since. not the best fix but its working better than i thought. i accidentally got an old patch in there, 0001-key-change.patch can be deleted.

### CODE REVIEW

**Comment by wabkia** [comment]
could i have changed the 'too close' skip into a deletion for waypoints or as a check before adding into the timeline? yes sounds great right? too bad i was annoyed and rewrote it to use RMB to set the waypoint. its working splendidly and also i can incorporate it back into the auto waypoint loop as a way to add more granular control. i probably need to add it back anyway because anyone with a fishy file previous to the change will be less optimized tho i can probably find a way around that also.

i learned a lot since yesterday and i realized that working with the human jitter, while an excellent idea, was covering up all sorts of issues. i tracked em down and have included them here :)

File: fishy/engine/fullautofisher/engine.py [hunk 1]
```diff
@@ -88,20 +91,21 @@ def run(self):
             self.window.crop = get_qr_location(self.window.get_capture())
             if self.window.crop is None:
                 log_raise("FishyQR not found, attempting rotate_back")
-                self.rotate_back()
+                # self.rotate_back()
```
Comment (adsau59) [inline fishy/engine/fullautofisher/engine.py:94]: dont keep commented code

File: fishy/engine/fullautofisher/engine.py [hunk 2]
```diff
@@ -136,39 +140,46 @@ def get_coords(self):
...
+        while FishingMode.CurrentMode not in cancel_states:
             print(f"walking for {walking_time}")
             kb.press(fishing_event.FishEvent.walk_key)
             time.sleep(walking_time)
             kb.release(fishing_event.FishEvent.walk_key)
             print("done")
             return True
         else:
-            return False
+            pass
```
Comment (adsau59) [inline fishy/engine/fullautofisher/engine.py:178]: looks like  you have inverted the if statement, ignore the comment on old commit

File: fishy/engine/fullautofisher/engine.py [hunk 3]
```diff
@@ -150,15 +150,15 @@ def move_to(self, target) -> bool:
...
-            pass
+        # if dist < 5e-05 and self.scan_counter <= 4:
+        # # if dist < 5e-05:
+        #     print("distance very small skipped {} move".format(self.scan_counter + 1))
+        #     self.scan_counter += 1
+        #     return True
+        # else:
+        #     logging.info("move_to() not skipped")
+        #     self.scan_counter = 0
+        #     pass
```
Comment (adsau59) [inline fishy/engine/fullautofisher/engine.py:161]: dont keep commented code pls

**Comment by adsau59** [comment]
move waypoint changes to other PR and keep this one only for the fighting system

**Review by adsau59 — CHANGES_REQUESTED** [review]

**Comment by adsau59** [comment]
Hi, sorry for the super late review, was busy with my job and this slipped from my head, thank you for the PR... I have requested a few changes, pls resolve them if they sound ok

File: fishy/engine/fullautofisher/engine.py (hunk 3 above)
Comment (SemjonKerner) [reply fishy/engine/fullautofisher/engine.py:161]: +1

File: fishy/engine/fullautofisher/engine.py [hunk 4]
```diff
@@ -204,7 +204,7 @@ def rotate_to(self, target_angle, from_angle=None) -> bool:
         for _ in range(abs(rotate_times)):
             if FishingMode.CurrentMode not in cancel_states:
                 mse.move(sign(rotate_times) * FullAuto.rotate_by * -1, 0)
-                time.sleep(0.05)
+                time.sleep(0.005)
```
Comment (SemjonKerner) [inline fishy/engine/fullautofisher/engine.py:207]: 5ms is to quick to be a reasonable timeout.

File: fishy/engine/fullautofisher/mode/recorder.py [hunk 5]
```diff
@@ -34,16 +34,45 @@ def __init__(self, engine: 'FullAuto'):
...
+        #     if not coords:
+        #         log_raise("QR not found")
+
+        #     end = player.find_nearest(old_timeline, coords)
+        #     self.engine.move_to(end[2])
+        #     part1 = old_timeline[:start_from[0]]
+        #     part2 = old_timeline[end[0]:]
+        #     self.timeline = part1 + self.timeline + part2
+        if not coords:
+            logging.warning("QR not found, couldn't record hole")
```
Comment (SemjonKerner) [inline fishy/engine/fullautofisher/mode/recorder.py:67]: this warning is covered in the QR engine

**Comment by SemjonKerner** [comment]
I tried reviewing this, but it is stuffed.

You really should split this up into logical units and create multiple PRs. There is some useful stuff in here, that should probably go into main, but other stuff definitely blocks this PR from getting merged, sorry.

Gonna close this PR in a few weeks, if noone objects. I hope you will consider publishing the best bits as separate PRs.

**Review by SemjonKerner — CHANGES_REQUESTED** [review]
I tried reviewing this, but it is stuffed.

You really should split this up into logical units and create multiple PRs. There is some useful stuff in here, that should probably go into main, but other stuff definitely blocks this PR from getting merged, sorry.

Gonna close this PR in a few weeks, if noone objects. I hope you will consider publishing the best bits as separate PRs.

**Comment by adsau59** [comment]
closing due to inactivity

**Pull request closed without merging**